### DISCUSSION TITLE
Add AO3 metadata

### DISCUSSION THREAD

**Discussion by ghost** [bookwyrm-social/bookwyrm#2174] [Ideas]
Maybe you can [convince them](https://archiveofourown.org) to federate so we can see their metadata from other instances.

Otherwise, expand the database to include their metadata and scrape their website.

**Comment by mouse-reeve** [comment]
Thank you! I'm going to convert this to a discussion because it needs well, discussion, before it could be an actionable issue. I know fanfiction is sometimes seen as a less legitimate genre, so I want to name that tendency and try to discuss this as a potential feature without denigrating any particular type of writing. BookWyrm's main use case is tracking reading activity on novels published through traditional publishers -- that doesn't mean it should be limited to that or that that type of writing is better, but it is the disposition of the service. Also fwiw I'm not an AO3 user or a fanfic reader so I may have some wrong ideas or obvious gaps in knowledge about how this works.

So the questions I have are, given that BookWyrm can't be everything for everyone, should it try to explicitly support types of writing with very different publication patterns and metadata formats? And informing that question, how different is the metadata from traditionally published fiction?

I'd love to hear from you and others as well if this would be an exciting feature, how (if at all) people are currently using BookWyrm to talk about fanfic, and what an ideal implementation would look like.

**Reply by ghost** [reply]
The main point in favor of adding this feature I see, is that goodreads allows completed fanfiction works, and at the end of the day I think bookwyrm wants to be a substitute to goodreads and implement most of its features.

For me the perfect implementation would be to include in the database all the metadata that fanfiction.net and archiveofourown.org implement and just be able to scrape all metadata for a work by providing its link in one of those websites.

The arechiveofourown.org advanced search is perfect so it could be copied just as it is. I think this would be very useful even for novels published through traditional publishers.

If incomplete works were supported it needs to be kept in mind the fact that they can be updated.

The metadata that this sites offer that would be most important would be completion status, fandom, tags for fanfiction tropes, content rating, pairings orientation, whether it's a crossover, number of words. And then less importantly known characters, relationships, number of chapters. There is some metadata like number of likes, etc. that change and unless it was going to be updated periodically it shouldn't be added.

Taking into account that this works are curated by the authors and aren't always properly categorized, I think it would be best if there was a voting system that allowed users to vote up or down on everything so that only the things with more than 0 total votes would show up, ordered by total votes (positives minus negatives) without actually showing the count. There would be some things that if one shows up the rest can't show up like completion status, rating, etc. For example an author may have indicated a work as complete when it is actually incomplete so when users vote on the incomplete tag, if there are more total votes up for it than for the complete tag, it would show incomplete but it wouldn't show complete even if the total (positives minus negatives) was above 0.

**Comment by franbarra** [comment]
Disclaimer: I don't read fanfiction or related works.

Let's imagine fanfiction work "A", is added by User 1 to their shelves. One week later, "A" gets updated through AO3 and User 2 adds it to their shelves. Do these two users have the same work in their bookshelves? What I'm trying to say is: how would BookWyrm handle books (fanfiction) that get updated (frequently or rarely, doesn't matter), which doesn't happen with traditional books that go through a publisher?

Secondly, would adding another connector for AO3 make the site slower for users who don't care about fanfiction (this is a possibility since we would be adding another site to query for search)? Would the search of fanfiction be enabled in settings so as not to disturb other users?

Last of all, referring to the design, I agree that while BookWyrm shouldn't try to encompass every use case for everyone, I also understand that maybe there is an overlap between fanfiction readers and "regular" books?

**Comment by elizabethyalkut** [comment]
There is definitely an overlap between fanfiction readers and book readers. That said: AO3 authors are generally very upset when people added their works to goodreads; here's a [couple of tumblr threads](https://abiroux.tumblr.com/post/105290278153/why-its-bad-fandom-etiquette-to-put-other) [that express opinions](https://mittensmorgul.tumblr.com/post/670311002876805120/occasional-random-reminder-that-the-vast-majority) on it. I would recommend against implementing this feature unless you have a way to _remove_ AO3 works on request, and that request process should be pretty lightweight on the part of the person requesting removal.

**Reply by mouse-reeve** [reply]
Thank you! I didn't know this and I definitely wouldn't want to add works against an authors wishes.

**Reply by jaschaurbach** [reply]
This is one reason why I deleted my AO3-ID branch - not only was the code a mess, this is another reason why I chose to not deliver better code to have AO3 IDs.
